### ISSUE TITLE
Database is not cleaned after wrong screenshot name error

### ISSUE THREAD

**Issue by thanosbellos** [rspec/rspec-rails#1868]
Hello. As descibed in #1865 using master branch and system specs, if there is a failed system spec, the screenshot is not named correctly. 
![failed-system-specs](https://user-images.githubusercontent.com/425283/30247950-cc4722fa-9627-11e7-8f3d-16dc653de6f2.png). So there is an Errno::ENOENT error raised. This causes some major complication in the teardown process of the example.

As wee see from the trace, rspec-rails delegates teardown to ActionDispatch::SystemTesting::TestHelpers::SetupAndTeardown#after_teardown
```ruby
#::ActionDispatch::SystemTesting::TestHelpers::SetupAndTeardown#after_teardown
module ActionDispatch
  module SystemTesting
    module TestHelpers
      module SetupAndTeardown # :nodoc:
        DEFAULT_HOST = "http://127.0.0.1"

        def host!(host)
          super
          Capybara.app_host = host
        end

        def before_setup
          host! DEFAULT_HOST
          super
        end

        def after_teardown
          take_failed_screenshot
          Capybara.reset_sessions!
          super
        end
      end
    end
  end
end
```
_::ActionDispatch::SystemTesting::TestHelpers::SetupAndTeardown#after_teardown_ calls _#take_failed_screenshot_ that raises the error as mentioned in the other issue #1865.

**The biggest complication is not that we don't have a correct screenshot. Rest of the after_teardown is not executed, so the real teardown after each example is not really happening. i.e The db records that are created in each example are not destroyed, so in the next example there would exist old records.**

[Here](https://github.com/thanosbellos/myapp) is a simple app with minimal setup that reproduces this behaviour. There are two similar system specs
1. The first one creates 2 widget, and fails because it expects to find 3 of them in index page
2. The second one creates two widgets and should succeed trying to find 2 of them in index page
3. if the first spec fails, the second one fails because the first 2 widgets are still in the database and in index base there are a total 4 of them.

I know that fixing the screenshot name bug should fix that problem too. Should we also provide information to rails team so that they ensure that the real teardown is happening if an error is raised by take_failed_screenshot?

**Comment by JonRowe** [comment]
If the code thats error'ing and failing to handle that error is in Rails then yes, this should all be handled upstream

**Comment by samphippen** [comment]
Could you check out the branch in [this PR](https://github.com/rspec/rspec-rails/pull/1872) and see if it fixes the problem?

**Comment by thanosbellos** [comment]
@samphippen gonna try it right away :)

**Comment by thanosbellos** [comment]
i tested in the sample [app](https://github.com/thanosbellos/myapp) **Everything seems great.** Screenshot name is now what it is expected to be. Teardown works nicely. Thanks a lot for your help and work. Pretty sure now that we should bother rails team to ensure that the teardown method is being executed even if an error is raised during tha take_screenshot phase.

**Comment by JonRowe** [comment]
Fixed by #1865
